### DISCUSSION TITLE
Tag is blocked from environment protection rules

### DISCUSSION THREAD

**Discussion by dmikurube** [community/community#63890] [Actions]
### Select Topic Area

Bug

### Body

I have an Environment to release a Java package to Maven Central. It has been configured to trigger by pushing a tag (ex. `v0.9.0`). It had been working for months-years.

Today, when I pushed a tag `v0.9.0`, the Action got an error `Branch "v0.9.0" is not allowed to deploy to maven-central due to environment protection rules.` even though the Environment is configured with branch protection `v0.*`.

Several similar problems are reported in comments for another (maybe unrelated) discussion: https://github.com/orgs/community/discussions/39054#discussioncomment-6727766

**Comment by ajaykarthikr** [comment]
I'm facing similar issue since 14 August 2023. I am unable to deploy from a tag with branch protection rule. This is a very recent issue.

![image](https://github.com/community/community/assets/19517804/32d56816-a956-4cfa-9189-ce78047cab9b)

Protection Rule:
![image](https://github.com/community/community/assets/19517804/c031445e-0d9e-498a-929b-cc56b3b8946c)

**Reply by fesaab** [reply]
Same issue here, started this week..

![image](https://github.com/community/community/assets/2200538/0c27a810-3d07-4890-bc68-40f6c2f9c749)

Environment deployment branches:
![image](https://github.com/community/community/assets/2200538/c77ae855-ee47-4ac7-820e-e5dc082ab9b7)

**Reply by evgfilim1** [reply]
Same (posted in another discussion)
https://github.com/orgs/community/discussions/39054#discussioncomment-6727766

**Comment by Jaber-Saed** [comment]
The error message you're encountering, "Branch 'v0.9.0' is not allowed to deploy to maven-central due to environment protection rules," suggests that there might be an issue with your GitHub Actions workflow configuration or the branch protection settings. Let's go through some steps to troubleshoot and resolve this issue:

    Verify Environment Protection Rules:
    Double-check your environment protection rules to ensure that the pattern v0.* is correctly configured to allow deployments. Make sure that the environment protection rules match the tag pattern you're pushing.

    Check Branch Protection Settings:
    Ensure that the branch protection settings for the repository are correctly configured to allow deployments from the v0.* tags. In the repository settings, go to "Branches" or "Code Protection" and check if there are any rules that might be conflicting with your intended deployment.

    Syntax and Wildcards:
    Verify that you're using the correct syntax and wildcards in both your workflow configuration and branch protection settings. GitHub's wildcards might be different from regular expressions, so be sure to follow the correct pattern matching rules.

    Case Sensitivity:
    Be aware that GitHub's tag matching can be case-sensitive. Ensure that the case of the tag in your tag push matches the case specified in your environment protection rules.

    Scope of Protection Rules:
    Make sure that the protection rules are applied to the correct branch(es). If you're pushing a tag, the protection rules should be applied to the branch that the tag is based on.

    Workflow Configuration:
    Review your GitHub Actions workflow configuration to ensure that it's correctly configured to trigger on tag pushes and to deploy to the environment. Double-check the workflow YAML file to see if there are any discrepancies.

    Permissions and Access:
    Ensure that your GitHub Actions workflow has the necessary permissions to deploy to the Maven Central repository. Check for any access issues that might be preventing the deployment.

    GitHub Support:
    If the issue persists and you've checked all the settings, consider reaching out to GitHub Support or the GitHub Community for further assistance. They can help you troubleshoot and resolve any underlying issues.

Remember that GitHub's features and behavior might have evolved since my last update in September 2021, so be sure to refer to the latest GitHub documentation and resources for the most up-to-date information and solutions.

**Reply by ajaykarthikr** [reply]
This is something that was working for me for months. This error came just recently. Nothing has been changed either in settings or GitHub action setup.

**Reply by dmikurube** [reply]
Same here. It's not a configuration issue.

**Reply by ivaivayl-flutterint** [reply]
Same here. With us, previous version was deployed successfully and this one is failing from today.
Check also https://github.com/orgs/community/discussions/39054... I've left my comment there with snapshot

**Reply by evgfilim1** [reply]
Looks like ChatGPT's answer :thinking:

**Comment by milotoor** [comment]
Recommend following along in another discussion on the same topic: #62991

**Reply by dmikurube** [reply]
Good to hear that! I was unable to find it.
